pkg/kv/kvserver/replica_application_result.go

```diff
@@ -211,30 +240,7 @@ func (r *replicaReproposer) newNotLeaseHolderError(msg string) *kvpb.NotLeaseHol
 func tryReproposeWithNewLeaseIndex(
 	ctx context.Context, cmd *replicatedCmd, r reproposer,
 ) *kvpb.Error {
-	// Note that we don't need to validate anything about the proposal's
-	// lease here - if we got this far, we know that everything but the
-	// index is valid at this point in the log.
 	p := cmd.proposal
-	if p.applied || p.Supersedes(cmd.Cmd.MaxLeaseIndex) {
-		// If the command associated with this rejected raft entry already applied
-		// then we don't want to repropose it. Doing so could lead to duplicate
-		// application of the same proposal. (We can see hit this case if an application
-		// batch contains multiple copies of the same proposal, in which case they are
-		// all marked as local, the first one will apply (and set p.applied) and the
-		// remaining copies will hit this branch).
-		//
-		// Similarly, if the proposal associated with this rejected raft entry is
-		// superseded by a different (larger) MaxLeaseIndex than the one we decoded
-		// from the entry itself, the command must have already passed through
-		// tryReproposeWithNewLeaseIndex previously (this can happen if there are
-		// multiple copies of the command in the logs; see
-		// TestReplicaRefreshMultiple). We must not create multiple copies with
-		// multiple lease indexes, so don't repropose it again. This ensures that at
-		// any time, there is only up to a single lease index that has a chance of
-		// succeeding in the Raft log for a given command.
-		return nil
```

Comment:
This is the `return nil` that would then set the caller's `pErr` to `nil`.

Reply:
There is another `return nil` below btw. Good that there its meaning is less overloaded now.